app/src/main/java/com/owncloud/android/datamodel/ThumbnailsCacheManager.java

```
@@ -289,7 +289,7 @@ protected Bitmap doInBackground(Object... params) {
             file = (OCFile) params[0];
```

Thanks for the fast response. It's working now like master, but file can be null, because **_Object... params_** maybe not OCFile, maybe empty OCFile array etc.

Could you add those null checks and type checks inside doInBackground function? 

```
            if (params != null && params.length > 0 && params[0] instanceof OCFile downloadedFile) {
                file = downloadedFile;
            } else {
                Log_OC.d(TAG, "Downloaded file is null");
                return null;
            }

            if (file.getRemoteId() == null || !file.isPreviewAvailable()) {
                Log_OC.d(TAG, "File cannot be previewed");
                return null;
            }
```

---

Sure thing, will do!

---

i added the check on params but inverted what you asked to implement a fail fast principle.

then the check you asked to add on the remote id and preview available was already "implemented" so i simply added a log when it enters the else statement.
Because if there is no remote id and the preview is false it will enter that else statement instead of trying to get from the cache or server

---

I've also just pushed the extraction in two separate methods!